### ISSUE TITLE
os x build issues

### ISSUE THREAD

**Issue by Eriken** [monostream/tifig#8]
Hello.

I am trying to build on os x 10.13.1 (high sierra) and I am getting some errors:
```
Eriks-MBP:build eriks$ cmake ..
-- The C compiler identification is AppleClang 9.0.0.9000038
-- The CXX compiler identification is AppleClang 9.0.0.9000038
-- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/cc
-- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/c++
-- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at /usr/local/Cellar/cmake/3.9.0/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find LibAV (missing: LIBAV_INCLUDE_DIR) (found version "12.2")
Call Stack (most recent call first):
  /usr/local/Cellar/cmake/3.9.0/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:377 (_FPHSA_FAILURE_MESSAGE)
  FindLibAV.cmake:220 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  CMakeLists.txt:9 (find_package)


-- Configuring incomplete, errors occurred!
See also "/Users/eriks/Documents/tifig/build/CMakeFiles/CMakeOutput.log".
```

I have installed both libav and vips through homebrew.

**Comment by Eriken** [comment]
I dowloaded libav and built it from source. configuration finished but the build process failed at the linking stage:

```
[ 95%] Linking CXX executable tifig
Undefined symbols for architecture x86_64:
  "_CVPixelBufferGetPixelFormatType", referenced from:
      _vda_decoder_callback in libavcodec.a(vda_h264.o)
  "_CVPixelBufferRelease", referenced from:
      _vda_h264_uninit in libavcodec.a(vda_h264.o)
      _ff_vda_output_callback in libavcodec.a(vda_h264.o)
      _release_buffer in libavcodec.a(vda_h264.o)
  "_CVPixelBufferRetain", referenced from:
      _vda_decoder_callback in libavcodec.a(vda_h264.o)
      _ff_vda_output_callback in libavcodec.a(vda_h264.o)
  "_VDADecoderCreate", referenced from:
      _ff_vda_create_decoder in libavcodec.a(vda_h264.o)
      _ff_vda_default_init in libavcodec.a(vda_h264.o)
  "_VDADecoderDecode", referenced from:
      _vda_old_h264_end_frame in libavcodec.a(vda_h264.o)
      _vda_h264_end_frame in libavcodec.a(vda_h264.o)
  "_VDADecoderDestroy", referenced from:
      _ff_vda_destroy_decoder in libavcodec.a(vda_h264.o)
  "_VDADecoderFlush", referenced from:
      _vda_old_h264_end_frame in libavcodec.a(vda_h264.o)
      _vda_h264_end_frame in libavcodec.a(vda_h264.o)
  "_avresample_alloc_context", referenced from:
      _opus_decode_init in libavcodec.a(opusdec.o)
  "_avresample_close", referenced from:
      _opus_decode_packet in libavcodec.a(opusdec.o)
      _opus_decode_flush in libavcodec.a(opusdec.o)
  "_avresample_convert", referenced from:
      _opus_decode_packet in libavcodec.a(opusdec.o)
  "_avresample_free", referenced from:
      _opus_decode_close in libavcodec.a(opusdec.o)
  "_avresample_is_open", referenced from:
      _opus_decode_packet in libavcodec.a(opusdec.o)
  "_avresample_open", referenced from:
      _opus_decode_packet in libavcodec.a(opusdec.o)
  "_compress", referenced from:
      _encode_strip in libavcodec.a(tiffenc.o)
  "_compress2", referenced from:
      _flashsv_encode_frame in libavcodec.a(flashsvenc.o)
  "_crc32", referenced from:
      _encode_frame in libavcodec.a(pngenc.o)
  "_deflate", referenced from:
      _flashsv_decode_frame in libavcodec.a(flashsv.o)
      _encode_frame in libavcodec.a(pngenc.o)
      _encode_frame in libavcodec.a(lclenc.o)
      _encode_frame in libavcodec.a(zmbvenc.o)
  "_deflateBound", referenced from:
      _flashsv_decode_frame in libavcodec.a(flashsv.o)
      _encode_frame in libavcodec.a(pngenc.o)
      _encode_frame in libavcodec.a(lclenc.o)
  "_deflateEnd", referenced from:
      _flashsv_decode_frame in libavcodec.a(flashsv.o)
      _flashsv_encode_end in libavcodec.a(flashsvenc.o)
      _encode_frame in libavcodec.a(pngenc.o)
      _encode_end in libavcodec.a(lclenc.o)
      _encode_end in libavcodec.a(zmbvenc.o)
  "_deflateInit2_", referenced from:
      _encode_frame in libavcodec.a(pngenc.o)
  "_deflateInit_", referenced from:
      _flashsv_decode_frame in libavcodec.a(flashsv.o)
      _encode_init in libavcodec.a(lclenc.o)
      _encode_init in libavcodec.a(zmbvenc.o)
  "_deflateReset", referenced from:
      _encode_frame in libavcodec.a(lclenc.o)
      _encode_frame in libavcodec.a(zmbvenc.o)
  "_inflate", referenced from:
      _flashsv_decode_frame in libavcodec.a(flashsv.o)
      _zlib_decomp in libavcodec.a(lcldec.o)
      _decode_frame in libavcodec.a(pngdec.o)
      _decode_frame in libavcodec.a(tiff.o)
      _decode_frame in libavcodec.a(tscc.o)
      _zerocodec_decode_frame in libavcodec.a(zerocodec.o)
      _decode_frame in libavcodec.a(zmbv.o)
      ...
  "_inflateEnd", referenced from:
      _flashsv_decode_init in libavcodec.a(flashsv.o)
      _flashsv_decode_end in libavcodec.a(flashsv.o)
      _flashsv2_decode_end in libavcodec.a(flashsv.o)
      _decode_end in libavcodec.a(lcldec.o)
      _decode_frame in libavcodec.a(pngdec.o)
      _decode_frame in libavcodec.a(tiff.o)
      _decode_end in libavcodec.a(tscc.o)
      ...
  "_inflateInit_", referenced from:
      _flashsv_decode_init in libavcodec.a(flashsv.o)
      _decode_init in libavcodec.a(lcldec.o)
      _decode_frame in libavcodec.a(pngdec.o)
      _decode_frame in libavcodec.a(tiff.o)
      _decode_init in libavcodec.a(tscc.o)
      _zerocodec_decode_init in libavcodec.a(zerocodec.o)
      _decode_init in libavcodec.a(zmbv.o)
      ...
  "_inflateReset", referenced from:
      _flashsv_decode_frame in libavcodec.a(flashsv.o)
      _zlib_decomp in libavcodec.a(lcldec.o)
      _decode_frame in libavcodec.a(tscc.o)
      _zerocodec_decode_frame in libavcodec.a(zerocodec.o)
      _decode_frame in libavcodec.a(zmbv.o)
  "_inflateSync", referenced from:
      _flashsv_decode_frame in libavcodec.a(flashsv.o)
  "_kCVPixelBufferIOSurfacePropertiesKey", referenced from:
      _ff_vda_create_decoder in libavcodec.a(vda_h264.o)
      _ff_vda_default_init in libavcodec.a(vda_h264.o)
  "_kCVPixelBufferPixelFormatTypeKey", referenced from:
      _ff_vda_create_decoder in libavcodec.a(vda_h264.o)
      _ff_vda_default_init in libavcodec.a(vda_h264.o)
  "_kVDADecoderConfiguration_Height", referenced from:
      _ff_vda_create_decoder in libavcodec.a(vda_h264.o)
      _ff_vda_default_init in libavcodec.a(vda_h264.o)
  "_kVDADecoderConfiguration_SourceFormat", referenced from:
      _ff_vda_create_decoder in libavcodec.a(vda_h264.o)
      _ff_vda_default_init in libavcodec.a(vda_h264.o)
  "_kVDADecoderConfiguration_Width", referenced from:
      _ff_vda_create_decoder in libavcodec.a(vda_h264.o)
      _ff_vda_default_init in libavcodec.a(vda_h264.o)
  "_kVDADecoderConfiguration_avcCData", referenced from:
      _ff_vda_create_decoder in libavcodec.a(vda_h264.o)
      _ff_vda_default_init in libavcodec.a(vda_h264.o)
  "_uncompress", referenced from:
      _decode_frame in libavcodec.a(cscd.o)
      _decode_frame in libavcodec.a(dxa.o)
      _decode_block in libavcodec.a(exr.o)
      _g2m_decode_frame in libavcodec.a(g2meet.o)
      _rscc_decode_frame in libavcodec.a(rscc.o)
      _screenpresso_decode_frame in libavcodec.a(screenpresso.o)
      _svq3_decode_init in libavcodec.a(svq3.o)
      ...
     (maybe you meant: _ff_snappy_peek_uncompressed_length, _ff_snappy_uncompress , _ff_lzf_uncompress )
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [tifig] Error 1
make[1]: *** [CMakeFiles/tifig.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by monostefan** [comment]
That's wierd, it works on all our high sierra machines. Can you please check with the current master branch, we just reworked and simplified the way our CMakeLists.txt finds LibAV. It might the job just fine now :)

If not, can you please run `brew info libav` and copy the output, then try `brew upgrade libav` and see if that solves your issue.

**Comment by monostefan** [comment]
Your second post came than I could answer the first one :). After you built from source, does `pkg-config libavcodec --libs` point to the directory where you built libav?

**Comment by Eriken** [comment]
After alot of back and forth I finally succeeded. The problem is that the libav package from homebrew does not include development files, only binaries. In order get them you need to install the ffmpeg package:
`brew install ffmpeg --with-tools --with-x265`

**Comment by monostefan** [comment]
Ah ok, thanks, that means that we linked against ffmpeg libraries on osx the whole time and didn't even realize it 😏. The problem here is that both LibAV and ffmpeg provide the same libraries  (`libavcodec`, `libswscale`, ...), which are basically interchangeable. I fixed the build instructions for osx based on your findings. The flag `--with-x265` shouldn't be necessary since it's just an encoder, we only need HEVC decoder from ffmpeg and that is included in the basic install. Thanks for your help!
